src/components/AreaController/AreaController.js

```diff
@@ -0,0 +1,57 @@
+import {fetchWatersheds} from '../../data-services/regions.js'
+import AreaSelector from '../AreaSelector/AreaSelector.js'
+import React, {useState} from 'react';
+import {findIndex} from 'lodash';
+
+function AreaController({onChangeRegionName, onChangeRegionBoundary}) {
+
+  const [regionNames, setRegionNames] = useState([]);
+  const [regionBoundaries, setRegionBoundaries] = useState([]);
+  const [currentRegionName, setCurrentRegionName] = useState("No Region Selected");
+  const [currentRegionBoundary, setCurrentRegionBoundary] = useState(null);
+  const [regionAreas, setRegionAreas] = useState();
+  const [currentRegionArea, setCurrentRegionArea] = useState();
```

Comment:
Need to bundle region boundaries, area, and names into a single data structure, especially as we want to add a lot more region metadata,

Reply:
Totally agree. See my global comment in this review.